lib/ir_pagerduty.rb

```diff
@@ -0,0 +1,19 @@
+require 'pagerduty/full'
+
+class IRPagerduty < PagerDuty::Full
+  def initialize
+    @apikey = SupportApp.pager_duty_token
+    @subdomain = SupportApp.pager_duty_subdomain
+  end
+
+  def fetch_json(path, params)
+    res = self.api_call(path, params)
+
+    case res
+      when Net::HTTPSuccess
+        JSON.parse(res.body)
+      else
+        res.error!
+    end
+  end
```

Comment:
this code is repeated in every method of the PagerDuty gems  nested classes:
```ruby
    case res
      when Net::HTTPSuccess
        JSON.parse(res.body)
      else
        res.error!
    end
```
I figured it would be best not to have to repeat it in the other current or future pager duty calls

Also, I thought it would be better to only reference the pager duty token/subdomain from one place rather that having to pass it through each time it is instantiated.

Happy to change though if you think it is better.